app/controllers/individual/UsingSoftwareController.scala

```diff
@@ -88,12 +89,21 @@ class UsingSoftwareController @Inject()(usingSoftware: UsingSoftware,
           }, yesNo =>
           sessionDataService.saveSoftwareStatus(yesNo) map {
             case Left(_) => throw new InternalServerException("[UsingSoftwareController][submit] - Could not save using software answer")
-            case Right(_) => {
-              if (isEnabled(PrePopulate)) Redirect(controllers.individual.tasklist.taxyear.routes.WhatYearToSignUpController.show())
-              else Redirect(controllers.individual.routes.WhatYouNeedToDoController.show)
-            }
+            case Right(_) => Redirect(getRedirect(yesNo))
           }
       )
   }
 
+  private def getRedirect(selectedOption: YesNo): String = {
+    if (isEnabled(PrePopulate)) {
+      selectedOption match {
+        case Yes => controllers.individual.tasklist.taxyear.routes.WhatYearToSignUpController.show().url
+        case No => controllers.individual.routes.NoSoftwareController.show.url
+      }
+    }
+    else {
+      controllers.individual.routes.WhatYouNeedToDoController.show.url
+    }
+  }
```

Comment:
```suggestion
  private def getRedirect(selectedOption: YesNo): Call = {
    if (isEnabled(PrePopulate)) {
      selectedOption match {
        case Yes => controllers.individual.tasklist.taxyear.routes.WhatYearToSignUpController.show()
        case No => controllers.individual.routes.NoSoftwareController.show
      }
    } else {
      controllers.individual.routes.WhatYouNeedToDoController.show
    }
  }
```
Can simplify this a little